external/dhcpd/dhcpd_lwip.c

```diff
@@ -109,66 +174,86 @@
  ****************************************************************************/
 int dhcp_server_status(char *intf)
 {
-	struct netif *cur_netif;
-	cur_netif = netif_find(intf);
-	if (cur_netif == NULL) {
-		ndbg("[DHCPS] no network interface for dhcpd start\n");
-		return 0;
+	int ret = ERROR;
+	struct req_lwip_data req;
+
+	int sockfd = socket(AF_INET, SOCK_DGRAM, 0);
+	if (sockfd < 0) {
+		ndbg("socket() failed with errno: %d\n", errno);
+		return ret;
 	}
 
-	if (cur_netif->dhcps_pcb == NULL) {
-		ndbg("[DHCPS] DHCP server closed\n");
-		return 0;
-	} else {
-		ndbg("[DHCPS] DHCP server opened\n");
-		return 1;
+	memset(&req, 0, sizeof(req));
+	req.type = DHCPDSTATUS;
+	req.host_name = intf;
+
+	ret = ioctl(sockfd, SIOCLWIP, (unsigned long)&req);
+	if (ret == ERROR) {
+		ndbg("ioctl() failed with errno: %d\n", errno);
+		close(sockfd);
+		return ret;
 	}
+
+	ret = req.req_res;
+	close(sockfd);
+	return ret;
 }
 
 /****************************************************************************
  * Name: dhcps_server_start
  ****************************************************************************/
+
+
 int dhcp_server_start(char *intf, dhcp_sta_joined dhcp_join_cb)
 {
-	struct netif *cur_netif;
-	cur_netif = netif_find(intf);
-	if (cur_netif == NULL) {
-		ndbg("[DHCPS] no network interface for dhcpd start\n");
-		return -1;
-	}
-	if (dhcp_join_cb) {
-		if (dhcps_register_cb(dhcp_join_cb) != ERR_OK) {
-			ndbg("[DHCPS] link callback fail\n");
-			return -1;
-		}
-	}
+	dhcp_join_data_s *data = (dhcp_join_data_s *)malloc(sizeof(dhcp_join_data_s));
+	data->intf = (char *)malloc(strlen(intf)+1);
```

Comment:
numm clech is needed for these 2.

Reply:
I guess it is 'null check'. Applied. Thanks for the comment.